src/utils/hooks/useSubmitReviewNEW.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  useUpdateReviewMutation,
+  Decision,
+  ReviewPatch,
+  ReviewResponseStatus,
+  Trigger,
+} from '../generated/graphql'
+import { FullStructure } from '../types'
+
+// below lines are used to get return type of the function that is returned by useUpdateReviewMutation
+type UseUpdateReviewMutationReturnType = ReturnType<typeof useUpdateReviewMutation>
+type promiseReturnType = ReturnType<UseUpdateReviewMutationReturnType[0]>
+// hook used to submit review, , as per type definition below (returns promise that resolve with mutation result data)
+type UseSubmitReview = (
+  reviewId: number
+) => (structure: FullStructure, decision: Decision) => promiseReturnType
+
+type ConstructReviewPatch = (structure: FullStructure, decision: Decision) => ReviewPatch
+
+const useSubmitReview: UseSubmitReview = (reviewId) => {
+  const [updateReview] = useUpdateReviewMutation()
+
+  const constructReviewPatch: ConstructReviewPatch = (structure, decision) => {
+    const reviewResponses = Object.values(structure?.elementsById || []).filter(
+      (element) => element.thisReviewLatestResponse
+    )
+
+    const reviewResponsesPatch = reviewResponses.map(({ thisReviewLatestResponse }) => ({
+      patch: {
+        decision: thisReviewLatestResponse?.decision,
+        comment: thisReviewLatestResponse?.comment,
+        status: ReviewResponseStatus.Submitted,
+      },
+      id: Number(thisReviewLatestResponse?.id),
+    }))
+
+    const reviewDecision = structure.thisReview?.reviewDecision
+
+    const reviewDecisionPatch = {
+      id: Number(reviewDecision?.id),
+      patch: {
+        comment: reviewDecision?.comment || undefined,
+        decision: decision,
+      },
+    }
+
+    return {
+      trigger: Trigger.OnReviewSubmit,
+      reviewResponsesUsingId: {
+        updateById: reviewResponsesPatch,
+      },
+      reviewDecisionsUsingId: {
+        updateById: [reviewDecisionPatch],
```

Comment:
Why the `reviewDecisionPatch` needs to go with brackets and the `reviewResponsesPatch` doesn't?

Reply:
yeah that's a bit confusing, i've changed here: d6e976a74133b2ffefd5fb48cf1a7b562c095e54, renamed to be reviewResponsesPatches